File: 13. Exam preparation/Variant-2/Problem 1/Cakes-solution.js
function solve(params) {
  var count1,
    count2,
    count3,
    price,
    s = params[0],
    c1 = params[1],
    c2 = params[2],
    c3 = params[3],
    answer = 0;

  for (count1 = 0; count1 <= (s / c1) + 1; count1++) {
    for (count2 = 0; count2 <= (s / c2) + 1; count2++) {
      for (count3 = 0; count3 <= (s / c3) + 1; count3++) {
        price = (count1 * c1) + (count2 * c2) + (count3 * c3);
        if (price <= s) {
          // if (price >= answer) console.log(s - price, i1, i2, i3);
          answer = Math.max(answer, price);
        }
      }
    }
  }

  return answer;
}

File: Topics/13. Exam preparation/Variant-2/Problem 1/Cakes-solution.cs
using System;

internal class Cakes
{
    private static void Main()
    {
        var s = int.Parse(Console.ReadLine());
        var c1 = int.Parse(Console.ReadLine());
        var c2 = int.Parse(Console.ReadLine());
        var c3 = int.Parse(Console.ReadLine());
        var answer = 0;

        for (var count1 = 0; count1 <= (s / c1) + 1; count1++)
        {
            for (var count2 = 0; count2 <= (s / c2) + 1; count2++)
            {
                for (var count3 = 0; count3 <= (s / c3) + 1; count3++)
                {
                    var price = (count1 * c1) + (count2 * c2) + (count3 * c3);
                    if (price <= s)
                    {
                        answer = Math.Max(answer, price);
                    }
                }
            }
        }

        Console.WriteLine(answer);
    }
}
